module/Motion/LineTracing.h

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file   LineTracing.h
+ * @brief  ライントレース動作
+ * @author CHIHAYATAKU
+ */
+
+#ifndef LINE_TRACING_H
+#define LINE_TRACING_H
+
+#include "SpeedCalculator.h"
+#include "Motion.h"
+#include "Mileage.h"
+#include "Timer.h"
+#include "Pid.h"
+
+class LineTracing : public Motion {
+ public:
+  /**
+   * コンストラクタ
+   * @param _targetSpeed 目標速度 0~
+   * @param _targetBrightness 目標輝度 0~
+   * @param _pidGain PIDゲイン
+   * @param _isLeftEdge エッジの左右判定(true:左エッジ, false:右エッジ)
+   */
+  LineTracing(double _targetSpeed, int _targetBrightness, const PidGain& _pidGain,
+              bool& _isLeftEdge);
+
+  /**
+   * @brief ライントレースする
+   */
+  void run();
+
+  /**
+   * @brief ライントレースする際の事前条件判定をする
+   * @param targetSpeed 目標速度
+   * @note オーバーライド必須
+   */
+
+ protected:
+  virtual bool isMetPreCondition(double targetSpeed) = 0;
+
+  /**
+   * @brief ライントレースする際の継続条件判定をする　返り値がfalseでモーターが止まる
+   * @note オーバーライド必須
+   */
+  virtual bool isMetContinuationCondition() = 0;
+
+  /**
+   * @brief 実行のログを取る
+   */
+  virtual void logRunning();
```

Comment:
```suggestion
  virtual void logRunning() = 0;
```
純粋仮想関数は =0 が必要だと思うんですが、返り値の型が void ならいらないとかってありましたっけ？
他の方の意見募集中です🙏
参考文献:
https://cpp-lang.sevendays-study.com/ex-day6.html
https://www.ibm.com/docs/ja/i/7.4?topic=only-abstract-classes-c

Reply:
参考文献を見たところ，実装が省略されている純粋仮想関数は=0が必要なのは確認しましたが，非純粋仮想関数（不純仮想関数?）については特に言及されていないようでした．
見落としがあるかもしれませんが，言及されていないなら=0はいらないのではないでしょうか．

今回は，LineTracing.cppでlogRunning()の実装はしているので，=0は無くてもよいのではと思います．
他の方の意見も聞きたいです！

Reply:
実装ない奴だけでいいと思う
virtualつけてて動的バインディングだから継承後の関数にoverrideつければ問題ない